src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/ScalarWaveGr.hpp

```diff
@@ -0,0 +1,203 @@
+                        std::array<double, volume_dim> center,
+                        std::unique_ptr<MathFunction<1>> profile,
+                        const OptionContext& context = {}) noexcept
+      : flat_space_scalar_wave_data_(std::move(profile)),
+        background_gr_data_(mass, dimensionless_spin, center, context) {}
+
+  // Enable or disable this constructor based on ScalarFieldData
+  template <typename Dummy = ScalarWave::Solutions::PlaneWave<volume_dim>,
+            Requires<std::is_same<ScalarFieldData, Dummy>::value> = nullptr>
+  explicit ScalarWaveGr(double mass,
+                        std::array<double, volume_dim> dimensionless_spin,
+                        std::array<double, volume_dim> center,
+                        std::array<double, volume_dim> wave_vector,
+                        std::array<double, volume_dim> wave_center,
+                        std::unique_ptr<MathFunction<1>> profile,
+                        const OptionContext& context = {}) noexcept
+      : flat_space_scalar_wave_data_(wave_vector, wave_center,
+                                     std::move(profile)),
+        background_gr_data_(mass, dimensionless_spin, center, context) {}
+
+  explicit ScalarWaveGr(CkMigrateMessage* /*unused*/) noexcept {}
+
+  ScalarWaveGr() = default;
+  ScalarWaveGr(const ScalarWaveGr& /*rhs*/) = delete;
+  ScalarWaveGr& operator=(const ScalarWaveGr& /*rhs*/) = delete;
+  ScalarWaveGr(ScalarWaveGr&& /*rhs*/) noexcept = default;
+  ScalarWaveGr& operator=(ScalarWaveGr&& /*rhs*/) noexcept = default;
+  ~ScalarWaveGr() = default;
+
+  // Tags
+  template <typename DataType>
+  using spacetime_tags = typename BackgroundGrData::template tags<DataType>;
+  using tags = tmpl::append<spacetime_tags<DataVector>,
+                            tmpl::list<Pi, Phi<volume_dim>, Psi>>;
+
+  /// Retrieve spacetime variables
+  template <
+      typename DataType, typename Tag,
+      Requires<tmpl::list_contains_v<spacetime_tags<DataType>, Tag>> = nullptr>
+  tuples::TaggedTuple<Tag> variables(const tnsr::I<DataType, volume_dim>& x,
+                                     tmpl::list<Tag> /*meta*/) const noexcept {
+    constexpr double default_initial_time = 0.;
+    return {std::move(get<Tag>(background_gr_data_.variables(
+        x, default_initial_time, spacetime_tags<DataType>{})))};
+  }
+
+  /// Retrieve scalar wave variables
+  tuples::TaggedTuple<Pi> variables(const tnsr::I<DataVector, volume_dim>& x,
+                                    tmpl::list<Pi> /*meta*/) const noexcept;
+  tuples::TaggedTuple<Phi<volume_dim>> variables(
+      const tnsr::I<DataVector, volume_dim>& x,
+      tmpl::list<Phi<volume_dim>> /*meta*/) const noexcept {
+    constexpr double default_initial_time = 0.;
+    return {std::move(
+        get<ScalarWave::Phi<volume_dim>>(flat_space_scalar_wave_data_.variables(
+            x, default_initial_time,
+            tmpl::list<ScalarWave::Pi, ScalarWave::Phi<volume_dim>,
```

Comment:
just use a `tmpl::list<ScalarWave::Phi<volume_dim>>`

Reply:
[`PlaneWave`](https://github.com/sxs-collaboration/spectre/blob/develop/src/PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.hpp) and [`RegularSphericalWave`](https://github.com/sxs-collaboration/spectre/blob/develop/src/PointwiseFunctions/AnalyticSolutions/WaveEquation/RegularSphericalWave.hpp) dont yet have the overloads of `variables` needed to fetch one tag at a time. I am happy to fix that in a follow-up PR to this.